diff --git a/ioslaves-master/keystore/iosl_master_keystore/iosl_master_keystore.ino b/ioslaves-master/keystore/iosl_master_keystore/iosl_master_keystore.ino
index e0c875f..037231a 100644
--- a/ioslaves-master/keystore/iosl_master_keystore/iosl_master_keystore.ino
+++ b/ioslaves-master/keystore/iosl_master_keystore/iosl_master_keystore.ino
@@ -1,6 +1,6 @@
 /**
- *  For Arduino Mega with Atmel 24C64AN = 8192 bytes
- *  Can store a 512B index and 30x 256B keys
+ *  For Arduino Mega with Atmel 24C32 = 4096 bytes
+ *  Can store a 512B index and 14x 256B keys
  *
  *  0x0000 /
  *    |    | { [slave_id \0][keynum_uint8] } x ...
@@ -16,10 +16,10 @@
  *  .      .
  *  .      .
  *  .      \_
- *  0x1800 /
- *         \_  keynum=28 256B key
- *  0x1900 /
- *  0x1999 \_  keynum=29 256B key
+ *  0x0E00 /
+ *         \_  keynum=12 256B key
+ *  0x0F00 /
+ *  0x0FFF \_  keynum=13 256B key
  *
  */
 
@@ -34,11 +34,11 @@
 typedef unsigned int romaddr_t;
 namespace eeprom {
 	const uint8_t I2C_EEPROM_ADDR = 0x50;
-	const unsigned int SZ = 0x2000; //B
+	const unsigned int SZ = 0x1000; //B
 	const romaddr_t IDX_ADDR = 0x0;
 	const unsigned int IDX_SZ = 512; //B
 	const romaddr_t KEYS_BEG = 0x200;
-	const uint8_t KEYS_NUM = 30;
+	const uint8_t KEYS_NUM = 14;
 	const short WRITE_DELAY_MS = 3; //ms
 	byte i2c_eeprom_read_byte (int device_addr, romaddr_t data_addr);
 	void i2c_eeprom_write_byte (int device_addr, romaddr_t data_addr, byte b);
@@ -129,8 +129,10 @@ void setup () {
 	Serial.begin(9600);
 	bgrid::initGrid();
 	Wire.begin();
-	pinMode(30, OUTPUT);
-	digitalWrite(30, LOW);
+	pinMode(18, OUTPUT);
+	digitalWrite(18, LOW);
+	pinMode(17, OUTPUT);
+	digitalWrite(17, HIGH);
 	pinMode(PIN_LED_RED, OUTPUT);
 	pinMode(PIN_LED_GREEN, OUTPUT);
 	delay(100);
